DuckDuckGoTests/ContextualOnboardingNewTabDialogFactoryTests.swift

```diff
@@ -0,0 +1,120 @@
+
+class ContextualOnboardingNewTabDialogFactoryTests: XCTestCase {
+
+    var factory: NewTabDaxDialogFactory!
+    var mockDelegate: CapturingOnboardingNavigationDelegate!
+    var onDismissCalled: Bool!
+
+    override func setUp() {
+        super.setUp()
+        mockDelegate = CapturingOnboardingNavigationDelegate()
+        onDismissCalled = false
+        factory = NewTabDaxDialogFactory(delegate: mockDelegate)
+    }
+
+    override func tearDown() {
+        factory = nil
+        mockDelegate = nil
+        onDismissCalled = nil
+        super.tearDown()
+    }
+
+    func testCreateInitialDialogCreatesAnOnboardingTrySearchDialog() {
+        // Given
+        let homeDialog = DaxDialogs.HomeScreenSpec.initial
+
+        // When
+        let view = factory.createDaxDialog(for: homeDialog, onDismiss: {})
+        let host = UIHostingController(rootView: view)
+        XCTAssertNotNil(host.view)
+
+        // Then
+        let trySearchDialog = find(OnboardingTrySearchDialog.self, in: host)
+        XCTAssertNotNil(trySearchDialog)
+        XCTAssertTrue(trySearchDialog?.viewModel.delegate === mockDelegate)
+    }
+
+    func testCreateSubsequentDialogCreatesAnOnboardingTryVisitingSiteDialog() {
+        // Given
+        let homeDialog = DaxDialogs.HomeScreenSpec.subsequent
+
+        // When
+        let view = factory.createDaxDialog(for: homeDialog, onDismiss: {})
+        let host = UIHostingController(rootView: view)
+        XCTAssertNotNil(host.view)
+
+        // Then
+        let trySiteDialog = find(OnboardingTryVisitingSiteDialog.self, in: host)
+        XCTAssertNotNil(trySiteDialog)
+        XCTAssertTrue(trySiteDialog?.viewModel.delegate === mockDelegate)
+    }
+
+    func testCreateFinalDialogCreatesAnOnboardingFinalDialog() {
+        // Given
+        var onDismissedRun = false
+        let homeDialog = DaxDialogs.HomeScreenSpec.final
+        let onDimsiss = { onDismissedRun = true }
+
+        // When
+        let view = factory.createDaxDialog(for: homeDialog, onDismiss: onDimsiss)
+        let host = UIHostingController(rootView: view)
+        XCTAssertNotNil(host.view)
+
+        // Then
+        let finalDialog = find(OnboardingFinalDialog.self, in: host)
+        XCTAssertNotNil(finalDialog)
+        finalDialog?.highFiveAction()
+        XCTAssertTrue(onDismissedRun)
+    }
+
+    func testCreateAddFavoriteDialogCreatesAnContextualDaxDialog() {
+        // Given
+        let homeDialog = DaxDialogs.HomeScreenSpec.addFavorite
+
+        // When
+        let view = factory.createDaxDialog(for: homeDialog, onDismiss: {})
+        let host = UIHostingController(rootView: view)
+        XCTAssertNotNil(host.view)
+
+        // Then
+        let addFavoriteDialog = find(ContextualDaxDialog.self, in: host)
+        XCTAssertNotNil(addFavoriteDialog)
+        XCTAssertEqual(addFavoriteDialog?.message.string, homeDialog.message)
+    }
+
+
+    private func find<T: View>(_ type: T.Type, in root: Any) -> T? {
```

Comment:
very nice, might be an idea to write up how to test swift view creation and share it with the team?